src/util/component-types.ts

```diff
@@ -63,7 +64,11 @@ interface ICreateClassComponentSpec<P, S> extends React.Mixin<P, S> {
 	// TODO: improve these with a stricter type https://stackoverflow.com/a/54775885/895558
 	reducers?: { [K in keyof P]?: (arg0: S, ...args: any[]) => S };
 	selectors?: { [K in keyof P]?: (arg0: S) => any };
-	render?(this: { props: P }): React.ReactNode;
+	render?(
+		this: { props: P } & {
+			[k: string]: any;
```

Comment:
@jhsu this was needed cause folks can drop additional properties on the class definition that would show up on `this` it turns out.

Reply:
oh interesting, didn't know that